packages/client/src/runtime/core/types/exported/Result.ts

```diff
@@ -48,7 +48,16 @@ export type FluentOperation =
 export type Count<O> = { [K in keyof O]: Count<number> } & {}
 
 // prettier-ignore
-export type GetFindResult<P extends OperationPayload, A> =
+export type TrueKeys<T> = {
+  [K in keyof T]: T[K] extends true ? K : never
+}[keyof T]
+
+export type GetFindResult<P extends OperationPayload, A> = A extends { exclude: infer Exclusion }
+  ? Compute<Omit<GetSelectIncludeResult<P, A>, TrueKeys<Exclusion>>>
+  : GetSelectIncludeResult<P, A>
+
+// prettier-ignore
+export type GetSelectIncludeResult<P extends OperationPayload, A> =
```

Comment:
🤯

Reading through the [infer docs](https://www.typescriptlang.org/docs/handbook/2/conditional-types.html#inferring-within-conditional-types) but I really appreciate the explanation of:
```ts
type ElementType<A> = A extends Array<infer X>
    ? x
    : never
```
translating to 
```rs
match A {
    Array(x) => x,
    _ => never,
}
```